### ISSUE TITLE
Problem with host group intersection

### ISSUE THREAD

**Issue by jsmartin** [ansible/ansible#3605]
For some reason I can't get the group intersection to work.  As you
can see I have a CentOS group and a riakcs_cluster group, and then I
want to run some plays on the intersection of those two groups, but it
seems like the intersection results in only one host (10.42.0.6).
What am I doing wrong?

```
PLAY [all:!localhost] *********************************************************

GATHERING FACTS ***************************************************************
ok: [10.42.0.200]
ok: [10.42.0.100]
ok: [10.42.0.7]
ok: [10.42.0.6]
ok: [10.42.0.8]

TASK: [Create groups based on distribution] ***********************************
changed: [10.42.0.100] => {"changed": true, "groups": {"CentOS":
["10.42.0.100", "10.42.0.200", "10.42.0.6", "10.42.0.7",
"10.42.0.8"]}}

PLAY [localhost] **************************************************************

GATHERING FACTS ***************************************************************
ok: [127.0.0.1]

TASK: [debug msg="{{groups['CentOS']}}"] **************************************
ok: [127.0.0.1] => {"msg": "['10.42.0.6', '10.42.0.7', '10.42.0.8',
'10.42.0.200', '10.42.0.100']"}

TASK: [debug msg="{{groups['riakcs_cluster']}}"] ******************************
ok: [127.0.0.1] => {"msg": "['10.42.0.7', '10.42.0.6', '10.42.0.8']"}

PLAY [CentOS:&riakcs_cluster] *************************************************

TASK: [fetch yum repository] **************************************************
ok: [10.42.0.6] => {"changed": false, "dest":
"/tmp/basho-release-6-1.noarch.rpm", "gid": 0, "group": "root",
"item": "", "mode": "0644", "msg": "file already exists", "owner":
"root", "size": 11311, "state": "file", "uid": 0, "url":
"http://yum.basho.com/gpg/basho-release-6-1.noarch.rpm"}

TASK: [configure the basho repository] ****************************************
ok: [10.42.0.6] => {"changed": false, "item": "", "msg": "", "rc": 0,
"results": []}

TASK: [install riak, riak-cs] *************************************************
ok: [10.42.0.6] => (item=riak-1.3.1,riak-cs-1.3.1) => {"changed":
false, "item": "riak-1.3.1,riak-cs-1.3.1", "msg": "", "rc": 0,
"results": ["riak-1.3.1-1.el6.x86_64 providing riak-1.3.1 is already
installed", "riak-cs-1.3.1-1.el6.x86_64 providing riak-cs-1.3.1 is
already installed"]}
```

Relevant playbook:

```
- hosts: all:!localhost
  tasks:
  - name: Create groups based on distribution
    group_by: key={{ ansible_distribution }}

- hosts: localhost
  tasks:
  - debug: msg="{{ groups['CentOS'] }}"
  - debug: msg="{{ groups['riakcs_cluster'] }}"

- hosts: CentOS:&riakcs_cluster
  gather_facts: no
  sudo: True
  roles:
  - bootstrap/redhat
  - riak/redhat
  - riak/common
  - riakcs/redhat
  - riakcs/common
```

**Comment by mpdehaan** [comment]
Ok, in queue to reproduce.

**Comment by mpdehaan** [comment]
Can you also share your inventory file?

Thanks!

**Comment by jsmartin** [comment]
```
[haproxy]
10.42.0.100

[localhost]
127.0.0.1

[riak_cluster]
10.42.0.6 node_type=primary
10.42.0.7 node_type=middle
10.42.0.8 node_type=last

[riak_servers:children]
riak_cluster

[stanchion]
10.42.0.6

[s3clients:children]
localhost

[riakcs_cluster:children]
riak_cluster

[riakcs_servers:children]
riakcs_cluster
```

**Comment by jimi-c** [comment]
It appears this has been corrected in 1.3, so I'm going to close this. Let us know if you run into it again and thanks!
